src/Disks/ObjectStorages/MetadataStorageFromPlainObjectStorageOperations.cpp

```diff
@@ -383,4 +388,152 @@ void MetadataStorageFromPlainObjectStorageCopyFileOperation::undo(std::unique_lo
     object_storage->removeObjectIfExists(StoredObject(remote_path_to));
     path_map.removeFile(path_to);
 }
+
+
+MetadataStorageFromPlainObjectStorageMoveFileOperation::MetadataStorageFromPlainObjectStorageMoveFileOperation(
+    bool replaceable_,
+    std::filesystem::path path_from_,
+    std::filesystem::path path_to_,
+    InMemoryDirectoryPathMap & path_map_,
+    ObjectStoragePtr object_storage_)
+    : replaceable(replaceable_)
+    , path_from(path_from_)
+    , remote_path_from(object_storage_->generateObjectKeyForPath(path_from_, std::nullopt).serialize())
+    , path_to(path_to_)
+    , remote_path_to(object_storage_->generateObjectKeyForPath(path_to_, std::nullopt).serialize())
+    , path_map(path_map_)
+    , object_storage(object_storage_)
+{
+    {
+        auto tmp_path_from = path_to.string() + ".move_from." + getRandomASCIIString(16);
+        tmp_remote_path_from = object_storage->generateObjectKeyForPath(tmp_path_from, std::nullopt).serialize();
+    }
+    {
+        auto tmp_path_to = path_to.string() + ".move_to." + getRandomASCIIString(16);
+        tmp_remote_path_to = object_storage->generateObjectKeyForPath(tmp_path_to, std::nullopt).serialize();
+    }
+}
+
+void MetadataStorageFromPlainObjectStorageMoveFileOperation::execute(std::unique_lock<SharedMutex> & /*metadata_lock*/)
```

Comment:
I deleted the previous comment and read the code more thoroughly. This method looks good.